src/fidesops/api/v1/endpoints/saas_config_endpoints.py

```diff
@@ -0,0 +1,150 @@
+        saas_config=saas_config,
+        validation_details=SaaSConfigValidationDetails(
+            msg=None,
+        ),
+    )
+
+
+@router.patch(
+    SAAS_CONFIG,
+    dependencies=[Security(verify_oauth_client, scopes=[SAAS_CONFIG_CREATE_OR_UPDATE])],
+    status_code=200,
+    response_model=ConnectionConfigurationResponse,
+)
+def patch_saas_config(
+    saas_config: SaaSConfig,
+    db: Session = Depends(deps.get_db),
+    connection_config: ConnectionConfig = Depends(_get_connection_config),
+) -> ConnectionConfig:
+    """
+    Given a SaaS config element, update the corresponding ConnectionConfig object
+    or report failure
+    """
+    data = {"key": connection_config.key, "saas_config": saas_config.dict()}
+    logger.info(f"Starting upsert for SaaS config '{saas_config.fides_key}'")
+    return ConnectionConfig.create_or_update(db, data=data)
+
+
+@router.get(
+    SAAS_CONFIG,
+    dependencies=[Security(verify_oauth_client, scopes=[SAAS_CONFIG_READ])],
+    response_model=SaaSConfig,
+)
+def get_saas_config(
+    connection_config: ConnectionConfig = Depends(_get_connection_config),
+) -> SaaSConfig:
+    """Returns the SaaS config for the given connection config."""
+
+    logger.info(f"Finding SaaS config for connection '{connection_config.key}'")
+    saas_config = connection_config.saas_config
+    if not saas_config:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND,
+            detail=f"No SaaS config found for connection '{connection_config.key}'",
+        )
+    return connection_config.saas_config
+
+
+@router.delete(
+    SAAS_CONFIG,
+    dependencies=[Security(verify_oauth_client, scopes=[SAAS_CONFIG_DELETE])],
+    status_code=204,
+)
+def delete_saas_config(
+    db: Session = Depends(deps.get_db),
+    connection_config: ConnectionConfig = Depends(_get_connection_config),
+) -> None:
+    """Removes the SaaS config for the given connection config.
+    The corresponding dataset and secrets must be deleted before deleting the SaaS config"""
+
+    logger.info(f"Finding SaaS config for connection '{connection_config.key}'")
+    saas_config = connection_config.saas_config
+    if not saas_config:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND,
+            detail=f"No SaaS config found for connection '{connection_config.key}'",
+        )
+
+    fides_key = saas_config["fides_key"]
+    dataset = DatasetConfig.filter(
+        db=db,
+        conditions=(
+            (DatasetConfig.connection_config_id == connection_config.id)
+            & (DatasetConfig.fides_key == fides_key)
+        ),
+    ).first()
+
+    warnings = []
+    if dataset:
+        warnings.append(
+            f"Must delete the dataset with fides_key '{fides_key}' before deleting this SaaS config."
+        )
+    if connection_config.secrets:
+        warnings.append(
+            "Must clear the secrets from this connection config before deleting the SaaS config."
+        )
```

Comment:
```
# The secrets must be cleared since the SaaS config is used for validation and the secrets
# might not pass validation once a new SaaS config is added.
```